packages/perseus/src/widgets/__testdata__/interactive-graph.testdata.ts

```diff
@@ -6,1715 +6,451 @@ import type {LockedFunctionOptions} from "../interactive-graphs/interactive-grap
-                    coords: [
-                        [3.5, 2],
-                        [3.5, 5],
-                        [-0.5, 2],
-                    ],
-                    match: "congruent",
-                    snapTo: "sides",
-                },
-            },
-            alignment: "default",
-        },
-    },
-};
+export const polygonWithFourSidesSnappingQuestion: PerseusRenderer =
+    interactiveGraphQuestionBuilder()
+        .withContent(
+            "**Example of snapping to sides** \n[[\u2603 interactive-graph 1]] \n",
+        )
+        .withGridStep(0.5, 0.5)
+        .withSnapStep(0.25, 0.25)
+        .withTickStep(0.5, 0.5)
+        .withMarkings("none")
+        .withXRange(-1, 6)
+        .withYRange(-1, 6)
+        .withPolygon("sides", {
+            match: "congruent",
+            numSides: 4,
+            showSides: true,
+            showAngles: false,
+        })
+        .build();
 
-export const polygonQuestionDefaultCorrect: PerseusRenderer = {
-    content:
-        "**Drag the vertices of the triangle below to draw a right triangle with side lengths $3$, $4$, and $5$.** \n[[\u2603 interactive-graph 1]] \n",
-    images: {},
-    widgets: {
-        "interactive-graph 1": {
-            graded: true,
-            version: {
-                major: 0,
-                minor: 0,
-            },
-            static: false,
-            type: "interactive-graph",
-            options: {
-                rulerTicks: 10,
-                showProtractor: false,
-                graph: {
-                    showSides: true,
-                    showAngles: false,
-                    snapTo: "grid",
-                    type: "polygon",
-                },
-                snapStep: [0.25, 0.25],
-                labels: ["x", "y"],
-                step: [0.5, 0.5],
-                gridStep: [0.5, 0.5],
-                backgroundImage: {
-                    scale: 1,
-                    bottom: 0,
-                    url: "",
-                    height: 0,
-                    width: 0,
-                    left: 0,
-                },
-                range: [
-                    [-1, 6],
-                    [-1, 6],
-                ],
-                showRuler: false,
-                markings: "none",
-                showTooltips: false,
-                rulerLabel: "",
-                correct: {
-                    showSides: true,
-                    type: "polygon",
-                    coords: [
-                        [3.5, 2],
-                        [2.5, 4],
-                        [1.5, 2],
-                    ],
-                    snapTo: "grid",
-                },
-            },
-            alignment: "default",
-        },
-    },
-};
+export const polygonQuestionDefaultCorrect: PerseusRenderer =
+    interactiveGraphQuestionBuilder()
+        .withPolygon("grid", {
+            startCoords: [
+                // TODO: figure out why this test is off by 0.5.
```

Comment:
Please make a ticket for this to figure out in the future. This isn't a ship-blocker

Reply:
https://khanacademy.atlassian.net/browse/LEMS-2186